tools/rosbag/src/record.cpp

```diff
@@ -145,8 +145,12 @@ rosbag::RecorderOptions parseOptions(int argc, char** argv) {
         if (freq == 0)
         {
             opts.custom_record_freq.emplace(it->first.as<std::string>(), ros::Duration(-1));
-        } 
-        else 
+        }
+        else if (freq == -1)
+        {
+            opts.custom_record_freq.emplace(it->first.as<std::string>(), ros::Duration(-2));
+        }
+        else
```

Comment:
as suggested by @ChristofDubs in the other PR, you can do `opts.custom_record_freq.emplace(it->first.as<std::string>(), ros::Duration( freq > 0 ? (double(1)/freq) : (freq-1)));` but I think current implementation is more readable